src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1268,7 +1268,8 @@ final int queueSize() {
         final void push(ForkJoinTask<?> task, ForkJoinPool pool,
                         boolean internal) {
             int s = top, b = base, m, cap, room; ForkJoinTask<?>[] a;
-            if ((a = array) != null && (cap = a.length) > 0) { // else disabled
+            if ((a = array) != null && (cap = a.length) > 0 &&
+                task != null) {                             // else disabled
```

Comment:
Can `task` ever be null here? or is this check just to avoid subsequent implicit nullchecks?